src/kiwixapp.cpp

```diff
@@ -142,25 +142,19 @@ void KiwixApp::newTab()
 
 QString KiwixApp::findLibraryDirectory()
 {
-    auto currentDataDir = QString::fromStdString(kiwix::removeLastPathElement(kiwix::getExecutablePath()));
-    
-    if (isPortableMode())
-        return currentDataDir + QDir::separator() + "data";
-
-    // Check for default dataDirectory.
-    currentDataDir = getDataDirectory();
-    auto libraryFile = QFileInfo(currentDataDir, "library.xml");
+    auto dataDir = getDataDirectory();
+    auto libraryFile = QFileInfo(dataDir, "library.xml");
     if (libraryFile.exists())
-        return currentDataDir;
+        return dataDir;
```

Comment:
This will fail to immediately return the portable data directory in portable mode if for some reason the library.xml file is missing (as far as I understand that's what happens during the first launch). Though the continued execution of this function will most likely have the same intended effect it is better to change this piece of code to:

```c++
    auto dataDir = getDataDirectory();
    if (isPortableMode())
        return dataDir;
```

Reply:
Sorry to postpone this, but why would that be a beneficial change? Isn't that just unneeded code?

Based on the current commit's implementation, the function would work as intended on first run, by returning on the last `return`

Reply:
Mostly yes. But I am confident that you can imagine the unlikely circumstances under which the heuristics implemented by the code below that line may be triggered wrongfully. Therefore the early return is a safety measure.

Reply:
Fair point. I will proceed then.